apps/studio/components/interfaces/Settings/Database/UsePoolerCheckbox.tsx

```diff
@@ -62,136 +62,129 @@ export const UsePoolerCheckbox = ({
-                          <DropdownMenu open={open} onOpenChange={setOpen}>
-                            <DropdownMenuTrigger asChild>
-                              <Button
-                                type="outline"
-                                className="py-0.5 pr-1.5"
-                                iconRight={<IconChevronDown strokeWidth={1} />}
-                              >
-                                <span className="capitalize">Mode: {poolingMode}</span>
-                              </Button>
-                            </DropdownMenuTrigger>
-                            <DropdownMenuContent side="bottom" align="start" className="w-[280px]">
-                              <DropdownMenuItem
-                                key="session"
-                                className="flex items-center justify-between"
-                                onClick={() => onSelectOption('session')}
-                              >
-                                <span>Session mode</span>
-                                {poolingMode === 'session' && <IconCheck className="text-brand" />}
-                              </DropdownMenuItem>
+    <div className="flex flex-col gap-y-1">
+      <div className="flex gap-x-3 items-center">
+        <Checkbox_Shadcn_
+          id={`use-pooler-${id}`}
+          checked={checked}
+          onCheckedChange={() => onCheckedChange(!checked)}
+        />
+        <div className="-mt-[2px] flex flex-col gap-y-1 w-full">
+          <div className="flex items-center gap-x-4">
+            <div className="flex items-center gap-x-2">
+              <Label_Shadcn_ htmlFor={`use-pooler-${id}`} className="text-sm cursor-pointer">
+                Display connection pooler
+              </Label_Shadcn_>
+              {checked && (
+                <>
+                  <div className="flex items-center gap-x-2">
+                    {isLoading && <ShimmeringLoader className="w-[100px] py-2.5" />}
+                    {isSuccess && (
+                      <div className="flex items-center gap-x-1">
+                        <DropdownMenu open={open} onOpenChange={setOpen}>
+                          <DropdownMenuTrigger asChild>
+                            <Button
+                              type="outline"
+                              className="py-0.5 pr-1.5 capitalize"
+                              iconRight={<ChevronDown strokeWidth={1} />}
+                            >
+                              Mode: {poolingMode}
+                            </Button>
+                          </DropdownMenuTrigger>
+                          <DropdownMenuContent side="bottom" align="start" className="w-[280px]">
+                            <DropdownMenuRadioGroup
+                              value={poolingMode}
+                              onValueChange={(value) =>
+                                onSelectOption(value as 'session' | 'transaction')
```

Comment:
we not have actual type variable?

Reply:
non blocking

Reply:
actually good point haha i should just type the value instead

but the type of value from onValueChange is just string
![image](https://github.com/supabase/supabase/assets/19742402/c6eda49d-5725-4d53-9ea0-219cc7b56332)